as-developers/rd.md

```diff
@@ -0,0 +1,275 @@
+| Note        | In the context of this document, a discrete bulletin-style message posted to a Course               |
+| Students    | A group of users that are registered to Courses and can see the content posted therein              |
+
+
+### 1.4 Overview
+
+## 2 Overall Description
+
+### 2.1 Product Perspective
+
+The centralized course delivery system to be developed will serve as a replacement for the current Brightspace system at UVic. The perspective for this system is outlined below:
+
+- The system will replace Brightspace for managing course grades, schedules, assignments, lecture videos and pdfs, and other course deliverables.
+- The system will be accessed by teachers and students on a variety of electronic devices via their netlink ID and password on a single centralized webapp.
+
+### 2.2 Product Features
+
+The system will serve as a centralized platform for teachers to deliver course content and to collect assignment submissions from students. The system will have a dedicated page for each course a student is registered in. These pages will display lecture slides and assignments in reverse chronological order for ease of navigation.
+
+### 2.3 User Classes and Characteristics
+There will be two types of users using this system: the student and instructor. The student class will have view-only permissions for most of the objects except for uploading assignments. The instructor class will be able to view and modify the objects. 
+
+#### 2.3.1 Student Users
+Student users can view all the objects that have been granted the access from the instructor user. In addition, a student user can upload and change the files under the 'Assignment' tab under the Course Content section within the starting and ending time assigned by the instructor class. 
+
+#### 2.3.2 Instructor Users
+Instructor users will have the permission to modify all the objects. They will be able to upload files under the Course Content section and assign a starting and ending date to each object under the section. They will also be able to input grade numbers under the Grade section and assign a formula of how the weight achieved is automatically calculated. 
+
+### 2.4 Operating Environment
+- Hardware platform: The system can be accessed using common web browser apps on both PC's and portable devices. The web browser apps need to support basic HTML and JavaScript extensions.
+- Operating system and version
+ 
+### 2.5 Design and Implementation Constraints
+
+_Hardware constraints_
+
+Students generally have access to relatively recent hardware and modern browsers, so support for older browsers such as
+Internet Explorer is not necessary. Since we will rely on cloud services for hosting, we will not consider hardware
+requirements—this responsibility will be left to the cloud provider.
+
+_Interaction with Third-Party Applications_
+
+The software may make use of third-party software libraries for development purposes as long as there is no concern of
```

Comment:
If you'd like to suggest a change, hit the suggestion button above the comment box...
```suggestion
The software may make use of third-party software libraries for development purposes when there is no concern of
```